contrib-02-lhcb-openstack/Readme.md

```diff
@@ -0,0 +1,192 @@
+# LHCbSoftware on Openstack
+
+## Motivation
+
+The easiest way to to work in an LHCb Software Evironment is by logging in to an `lxplus` instance.
+From there one also has access to the [`AFS` file storage](http://information-technology.web.cern.ch/services/afs-service).
+
+However, working on `lxplus` comes with a few significant limitations:
+ * No super user access. This makes it hard to install new software.
+ * Shared resources with other users. Others might start resource-hungry processes on the same machine, therefore extending execution time and making it unpredicatable.
+
+Using a virtual machine on [`openstack.cern.ch`](https://openstack.cern.ch) solves these issues. It is possible to disposition a machine with up to 10 vCPUs and up to 20 GB memory. While vCPUs are not the same as actual, physical CPUs, this comes with with a significant improvement in performance and performance reliablity.
+However, the default VM images do not have the LHCb Software Environment installed, nor do they have access to AFS.
+
+This tutorial shows how one can easily deploy a machine on Openstack which offers the LHCb software environment and AFS access, using the CernVM virtual machine image.
+This can be done using the command line only. No step in the tutorial requires any GUI interaction.
+
+You can now deploy a powerful machine on openstack, run LHCb Software there while enjoying full super user access and access to AFS. You can take snapshots of your machines and if all goes awry, you can choose the nuclear option on your machine - and set up a new one.
+
+Especially useful for:
+  * Continous integration tests (CI)
+  * Profiling your algorithms.
+
+
+
+## Set up an Openstack instance with AFS and the LHCb Software environment
+
+
+This is a very basic tutorial, some steps described can be skipped if you have already done them.
+
+
+### Register for [OpenStack](https://openstack.cern.ch). 
+The way to register is described [here](https://clouddocs.web.cern.ch/clouddocs/tutorial_using_a_browser/subscribe_to_the_cloud_service.html) in the [OpenStack documentation](https://clouddocs.web.cern.ch/clouddocs/)
+
+### Add public key to Openstack
+Log in to lxplus,  create an ssh key-pair. Add your public key to to your [list of public keys on openstack](https://openstack.cern.ch/project/access_and_security/)
+
+
+### Download `openrc.sh`
+Download your personal OpenStack RC file. 
+You can find it on https://openstack.cern.ch/project/access_and_security/ if you click on the "API Access" index tab.  
+Save it as `openrc.sh`.
+
+### Set up your environment
+
+```
+source ~/openrc.sh
+export OS_CACERT=/etc/pki/tls/cert.pem
+```
+    
+### Upload [CERNVM](https://cernvm.cern.ch/) image
+
+Upload the CERNVM image to OpenStack (details [here](https://cernvm.cern.ch/portal/openstack) ) 
+    
+```
+wget http://cernvm.cern.ch/releases/production/cernvm-3.5.1.hdd
```

Comment:
I think you can do it. We have to investigate it, but see that you did `--is-public False`. Probably not doing that it would be possible to have a common image.